### ISSUE TITLE
`--stop-on-failure` no longer stops on errors

### ISSUE THREAD

**Issue by Jean85** [sebastianbergmann/phpunit#5145]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 10.x-dev
| PHP version         | 8.1
| Installation Method | Composer

#### Summary

`--stop-on-failure` no longer stops on errors (as it happened on 9.x),

I don't know if this is intended (and has to be documented as a BC in the Changelog) or if it's a bug...

**Comment by garak** [comment]
Sorry to comment on a closed issue, but I see that this bug is supposed to be fixed in version 10.0.4, but I still get it:
```bash
bin/phpunit --stop-on-failure

PHPUnit 10.0.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.1.1
Configuration: /user/myapp/phpunit.xml.dist

.....E.......................................................     61 / 61 (100%)

Time: 00:05.680, Memory: 72.50 MB

There was 1 error

```

**Comment by sebastianbergmann** [comment]
Quoting from the ChangeLog:

* The CLI test runner now only stops after a test errored when `--stop-on-error` or `--stop-on-defect` is used
* The CLI test runner now only stops after a test failed when `--stop-on-failure` or `--stop-on-defect` is used
* The CLI test runner now only stops after a test triggered a warning when `--stop-on-warning` or `--stop-on-defect` is used
* The CLI test runner now only stops after a test was marked as risky when `--stop-on-risky` or `--stop-on-defect` is used
* The CLI test runner now only stops after a test was skipped when `--stop-on-skipped` is used
* The CLI test runner now only stops after a test was marked as incomplete when `--stop-on-incomplete` is used

TL;DR: There is no bug here, this works as intended.

**Comment by federicotorrisi** [comment]
hello @sebastianbergmann ,
we are currently using v10.0.19 and phpunit is still not stopping using the flag `--stop-on-defect` when a warning is raised.

we also tried with dev-main 3a680a9 but still no luck.
any ideas?

**Comment by sebastianbergmann** [comment]
> Sorry to comment on a closed issue, but I see that this bug is supposed to be fixed in version 10.0.4, but I still get it:
> 
> ```shell
> bin/phpunit --stop-on-failure
> 
> PHPUnit 10.0.4 by Sebastian Bergmann and contributors.
> 
> Runtime:       PHP 8.1.1
> Configuration: /user/myapp/phpunit.xml.dist
> 
> .....E.......................................................     61 / 61 (100%)
> 
> Time: 00:05.680, Memory: 72.50 MB
> 
> There was 1 error
> ```

An error is not a failure, these are treated separately.

**Comment by sebastianbergmann** [comment]
> hello @sebastianbergmann , we are currently using v10.0.19 and phpunit is still not stopping using the flag `--stop-on-defect` when a warning is raised.
> 
> we also tried with dev-main [3a680a9](https://github.com/sebastianbergmann/phpunit/commit/3a680a9c0f2d0651cfe7c45cb0fcf5cd926d6617) but still no luck. any ideas?

Please open a new ticket and provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting. Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

According to this [test](https://github.com/sebastianbergmann/phpunit/blob/10.0/tests/end-to-end/cli/stop-on/stop-on-defect-for-warning.phpt), `--stop-on-defect` correctly aborts test execution after the first test that triggered a warning.

**Comment by Jean85** [comment]
Maybe the `failOnWarning` setting is involved in the issue? @federicotorrisi what's the value that you use on that?

**Comment by federicotorrisi** [comment]
@Jean85  `failOnWarning` is currently set to `true`
